trieste/models/gpflux/models.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import warnings
+
+import tensorflow as tf
+from gpflux.layers import GPLayer
+from gpflux.models import DeepGP
+from gpflux.models.deep_gp import sample_dgp
+
+from ...data import Dataset
+from ...types import TensorType
+from ...utils import jit
+from ..interfaces import TrainableProbabilisticModel
+from ..optimizer import Optimizer, TFOptimizer
+from .interface import GPfluxPredictor
+
+
+class VanillaDeepGP(GPfluxPredictor, TrainableProbabilisticModel):
+    """
+    A :class:`TrainableProbabilisticModel` wrapper for a GPflux :class:`~gpflow.models.DeepGP` with
+    only standard :class:`GPLayer`: this class does not support keras layers, latent variable
+    layers, etc.
+    """
+
+    def __init__(self, model: DeepGP, optimizer: Optimizer | None = None):
+        """
+        :param model: The underlying GPflux deep Gaussian process model.
+        :param optimizer: The optimizer with which to train the model. Defaults to
+            :class:`~trieste.models.optimizer.TFOptimizer` with :class:`~tf.optimizers.Adam` with
+            batch size 100.
+        """
+
+        if optimizer is None:
+            optimizer = TFOptimizer(tf.optimizers.Adam(), batch_size=100)
+
+        if not isinstance(optimizer, TFOptimizer):
+            raise ValueError("Optimizer must be a TFOptimizer for deep GPs")
+
+        super().__init__(optimizer)
+
+        if not all([isinstance(layer, GPLayer) for layer in model.f_layers]):
+            raise ValueError("`VanillaDeepGP` can only be built out of `GPLayer`")
+
+        for layer in model.f_layers:
+            if layer.whiten:
+                warnings.warn("Sampling cannot be currently used with whitening in layers")
+
+        self._model = model
+
+    def __repr__(self) -> str:
+        """"""
+        return f"VanillaDeepGP({self._model!r}, {self.optimizer!r})"
+
+    @property
+    def model(self) -> DeepGP:
+        return self._model
+
+    def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
+        sampler = sample_dgp(self.model)
+
+        @jit(apply=self.optimizer.compile)
```

Comment:
we can make it an empirical question, make it a parameter in the class and we can test on certain number of datasets and architechtures whether its useful?

Reply:
Is this referring to whether to compile?

Reply:
yes...